File: frontend/src/api/workflow.js
import http from './http'

export const getWorkflows = (params) => http.get('/workflows', { params })
export const getWorkflow = (id) => http.get(`/workflows/${id}`)
export const createWorkflow = (data) => http.post('/workflows', data)
export const updateWorkflow = (id, data) => http.put(`/workflows/${id}`, data)
export const deleteWorkflow = (id) => http.delete(`/workflows/${id}`)

export const getBizTypes = () => http.get('/biz-types')
export const createBizType = (data) => http.post('/biz-types', data)
export const deleteBizType = (id) => http.delete(`/biz-types/${id}`)

File: backend/routes/workflow.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

func workflowRoutes(rg *gin.RouterGroup) {
	rg.GET("/workflows", handlers.GetWorkflowTemplates)
	rg.GET("/workflows/:id", handlers.GetWorkflowTemplate)
	rg.POST("/workflows", handlers.CreateWorkflowTemplate)
	rg.PUT("/workflows/:id", handlers.UpdateWorkflowTemplate)
	rg.DELETE("/workflows/:id", handlers.DeleteWorkflowTemplate)

	// 业务类型
	rg.GET("/biz-types", handlers.GetBizTypes)
	rg.POST("/biz-types", handlers.CreateBizType)
	rg.DELETE("/biz-types/:id", handlers.DeleteBizType)

}
